src/useFocusContext.ts

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+
+export interface FocusContext<Element extends HTMLElement = HTMLElement> {
+   elements: Element[];
+   focusedIndex: number;
+   register: (element: Element) => void;
+   unregister: (element: Element) => void;
+   focusFirst: () => void;
+   focusLast: () => void;
+   focusNext: () => void;
+   focusPrevious: () => void;
+   focus: (index: number) => void;
+   blur: () => void;
+}
+
+export function useFocusContext<
+   Element extends HTMLElement = HTMLElement
+>(): FocusContext<Element> {
+   const [focusedIndex, setFocusedIndex] = React.useState(-1);
+   const [elements, setElements] = React.useState<Element[]>([]);
+
+   const register = React.useCallback((element: Element) => {
+      setElements((currentElements) => {
+         const isAlreadyInserted = currentElements.some((el) => el === element);
```

Comment:
I think this is equivalent. Any reason to use some?
```suggestion
         const isAlreadyInserted = currentElements.includes(element);
```